CRM/Contribute/BAO/ContributionSoft.php

```diff
@@ -307,31 +322,28 @@ public static function getSoftContribution($contributionID, $all = FALSE) {
       'pcp_personal_note',
     );
 
-    $query = '
-    SELECT ccs.id, pcp_id, cpcp.title as pcp_title, pcp_display_in_roll, pcp_roll_nickname, pcp_personal_note, ccs.currency as currency, amount, ccs.contact_id as contact_id, c.display_name, ccs.soft_credit_type_id
+    $query = "
+    SELECT ccs.id, pcp_id, ccs.contribution_id as contribution_id, cpcp.title as pcp_title, pcp_display_in_roll, pcp_roll_nickname, pcp_personal_note, ccs.currency as currency, amount, ccs.contact_id as contact_id, c.display_name, ccs.soft_credit_type_id
     FROM civicrm_contribution_soft ccs INNER JOIN civicrm_contact c on c.id = ccs.contact_id
     LEFT JOIN civicrm_pcp cpcp ON ccs.pcp_id = cpcp.id
-    WHERE contribution_id = %1;
-    ';
+    WHERE contribution_id IN (" . implode(',', $contributionIDs) . ")
```

Comment:
You can keep that parameter escaped by using `[implode(',', $contributionIDs), 'CommaSeparatedIntegers']` in `executeQuery`. I believe this code path is safe, but future ones might pass unvalidated values.

Reply:
@pfigel Thankyou, I couldn't see how to do it but I've done it now.